### PR TITLE
Add ListGroup and GroupOption component

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Add the `ListGroup` component.

#### How should this be manually tested?

[Workspace](https://lucas--checkoutio.myvtex.com/checkout/#/shipping).

This component is being used in the shipping step.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/ListGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+const ListGroup: React.FC = ({ children }) => {
+  return (
+    <div className="nr5 nr0-ns bt bb b--muted-4">
+      <div className="nl5 nl0-ns pa0" role="group">
```
Comment (kevinch) [inline react/ListGroup.tsx]: A `div` has no default padding so this class `pa0` might not be useful.

Also can't this inner div be merged with its parent?

Comment (kevinch) [reply react/ListGroup.tsx]: And why the negative margin?

Comment (lucasecdb) [reply react/ListGroup.tsx]: nice catch! I used to have a `ul` element here, so that's why I used the `pa0` class

Comment (lucasecdb) [reply react/ListGroup.tsx]: the negative margin is tricky, but basically its so I can have the width of the group to be 100% on mobile, but to be limited to the parent when not

Comment (lucasecdb) [reply react/ListGroup.tsx]: I'll see about merging these two divs

Comment (lucasecdb) [reply react/ListGroup.tsx]: the difference of the two divs being separated is subtle, but isn't what I want :-/

separated:

image

merged:

image

the only difference is that the outer border must go all the way to the right, but need the margin on the left

**Review by pedromtec — APPROVED** [review]
LGTM

**Review by thiagomurakami — APPROVED** [review]
lgtm

**Pull request merged**